### ISSUE TITLE
Replacing evil-leader

### ISSUE THREAD

**Issue by wpcarro** [noctuid/general.el#114]
I was trying to replace `evil-leader` with `general` to reduce my configuration's size. I'm surprised that I wasn't able to get it to work with the following code:

```elisp
(general-define-key
 :prefix "SPC"
 :states 'motion
 :keymaps 'override
 "f" #'wpc/find-file)
```

I was eventually able to get things working doing this:

```elisp
(general-create-definer wpc/leader-def
  :prefix "SPC")

(wpc/leader-def
 :keymaps 'normal
 "f"  #'wpc/find-file)
```

In the spirit of trying to learn more about `general.el`, can you help me debug the shortcomings of the first snippet?

**Comment by noctuid** [comment]
The initial example should be `:keymaps 'override :states 'normal`. Currently the override keymaps are only set to override evil for the exact state specified, so unless you are actually in motion state, the first example will have no effect. I can make this more clear in the documentation

**Comment by wpcarro** [comment]
@noctuid interesting, thanks. It's probably a shortcoming on my end. I just figured that `motion` was preferable to `normal`since it was a superset of normal. I guess my understanding of `motion` needs to improve.

**Comment by noctuid** [comment]
Normally keybindings made in motion state are inherited in other states. Keybindings made in normal state are inherited by other states as well excluding motion state, so unless you want to use motion state directly in some modes, you don't really need to bind keys in motion state. Personally, I prefer to always use normal state.

It might be possible to set things up so that your original example works, but there may be issues with precedence. I will look into this because ideally there should be some inheritance (e.g. you should potentially be able to use a key bound in normal state with `:keymaps 'override :states 'normal` in visual state as well).

**Comment by wpcarro** [comment]
Thanks for the explanation. That confirms some of my hunches. For the record, I would like to have access to most of my "leader" bindings from `motion` states. At least that's what I *think*... off the top of my head I can't think of any motion states that I use, but I was under the impression that evil designated that mode for some buffers. Will need to take a closer look... 

Either way, I will heed your advice about `normal` for now until I have a better grasp of what I'm looking for. As the maintainer of `general.el`, and after reading your write-ups on how KBDs work in Emacs, I'll defer to your expertise.

Thanks again!

**Comment by wpcarro** [comment]
As for your original advice about changing `motion` -> `normal`, I can't get the same behavior that I was getting with my custom definer.

Here's the code:

```elisp
(general-define-key
 :prefix "SPC"
 :states 'normal
 :keymaps 'override
 "f"  #'wpc/find-file
 )
```

**Comment by wpcarro** [comment]
What's interesting is that the problem seems to be coming from the presence of `:keymaps 'override`. Once I remove that, everything seems to work just fine.

**Comment by noctuid** [comment]
> I was under the impression that evil designated that mode for some buffers.

Yes, by default various read-only buffers default to motion state (see `evil-motion-state-modes`), but this is easily configurable. The reasoning is that motions (which are bound in motion state by default) are useful in read-only modes but insertion commands (which are bound in normal state) aren't. I don't think it makes sense to create/use a separate state for this purpose (e.g. evil-collection now uses normal state for read-only modes and just ignores insertion commands). In practice, you shouldn't really need to worry about the difference.

> What's interesting is that the problem seems to be coming from the presence of :keymaps 'override. Once I remove that, everything seems to work just fine.

Your example works fine for me. Have you enabled `(general-override-mode)`? And if you have, are you using the latest version of evil?

**Comment by wpcarro** [comment]
Ugh apologies, I didn't have `general-override-mode`. I've re-read your docs on the `override` keymap; that's totally my fault.

After running `(general-override-mode)` at startup, the bindings work as expected.

A couple of thoughts from debugging. 

1. Would be nice to have a warning when someone attempts to opt into `override` without having run `(general-override-mode)`. I imagine this being akin to `evil-collection` emitting warnings when `evil-want-integration` is `nil` but `evil-collection-init` is called.
2. Perhaps a section dedicated to replacing `evil-leader` and showing a direct comparison with an `evil-leader` setup and the alternative, `general` one. I could help with the latter more so than the former.

**Comment by noctuid** [comment]
`general-override-mode` is now automitcally enabled by default since most people probably won't want to toggle it off later.

As for your second point, if you think that the documentation for `general-create-definer` or `:prefix` is insufficient or that it could be more clearly linked to early on in the readme, then I can update the readme. For a direct comparison between using `evil-leader` and `general`, feel free to add an entry to the wiki.

**Comment by wpcarro** [comment]
@noctuid I think with `general-override-mode` enabled by default, that should clear up some misunderstandings. If you disagree, however, let me know, and I'll be happy to provide some code snippets about my porting process.
